dvc/stage.py

```diff
@@ -29,12 +29,16 @@ def __init__(self, fname, e):
 
 
 class StageFileDoesNotExistError(DvcException):
-    def __init__(self, fname):
+    def __init__(self, fname, from_checkout=False):
         msg = "'{}' does not exist.".format(fname)
 
         sname = fname + Stage.STAGE_FILE_SUFFIX
+        prefix = " Do you mean"
         if Stage.is_stage_file(sname):
-            msg += " Do you mean '{}'?".format(sname)
+            msg += "{} '{}'?".format(prefix, sname)
+            prefix = " Or perhaps"
+        if from_checkout:
```

Comment:
Could you please also add a unit test for it to `tests/unit/stage.py` that would check that with and without `from_checkout==True` and with and without `fname` being a dvc file it raises expected messages?

Reply:
Yes, thanks for the unit tests! :)

Good point! Yes, we should handle that case as well. Let's not create a yet another `if`, but rather catch both exceptions and just append a message about git branch. What do you think?